[local-authentication][iOS] allow for more flexibility in arguments

cruzach opened the pull request:
# Why

In attempting to repro [4799](https://github.com/expo/expo/issues/4799), I discovered that supplying only the `options.fallbackLabel` argument to `authenticateAsync` results in an error. To make this API a little more user-friendly, we should allow it to be called with no arguments, both arguments, or just one.

cruzach committed:
[docs] make ios simulator limitations clear (#5838)

Commit:
[docs] Remove SDK 30, 31 and 32 versions

Commit:
[docs] Update using-firebase.md

cruzach committed:
[local-authentication][iOS] allow for more flexibility in arguments

cruzach committed:
add default promptMessage string to docs

EvanBacon committed:
Added expo badge to the readme (#5826)

* Added expo badge to the readme

* Update README.md

EvanBacon committed:
Fix badge link

tsapeta committed:
[ios][et][template-files] Make client's podfile static (#5783)

# Why

So far we've been generating client's podfile in `et ios-generate-dynamic-macros` and that file was git-ignored. This way was causing a lot of issues with Podfile locks and Pods folder being out of sync after merging pull requests or rebasing branches.

# How

- Removed `/ios/Podfile` from `.gitignore`
- Replaced every template substitution in client's podfile template with the proper value and made it a static file. Most of those substitutions were already static or they differed only between SDK versions, but that's still fine.
- Two substitutions were more problematic - as they are being generated during versioning process. I decided to make a `PodfileHelpers.rb` script with some helpers that will be evaluating `ios/versioned-react-native/ABI*/{dependencies,postinstalls}.rb` scripts inside podfile. That way, we can still generate versioned parts of the podfile and automatically evaluate them when running `pod` commands.
- Removed `:inhibit_warnings => true` from all deps in podfile and used `inhibit_all_warnings!` at the top of the file with one small exception - unversioned unimodules will override that default setting with `:inhibit_warnings => false`. I also added a possibility to `r-n-u` to define custom pod flags: see https://github.com/unimodules/react-native-unimodules/commit/185a86a265d57f5e0f6b90401975ccf20b17aba3.
- `et ios-generate-dynamic-macros` no longer generates client's podfile.
- Updated `et add-sdk` & `et rm-sdk` to use `dependencies.rb` and `postinstalls.rb` scripts under `versioned-react-native/ABI*` folders instead of `template-files`. In fact, they are not templates now.
- Moved some dependencies and their version requirements from client's podfile to the unimodules that depend on them. For example, client doesn't really need to specify version requirement for `Amplitude` pod as it isn't used by the client directly, but by `expo-analytics-amplitude` unimodule. This also fixes some incompatibility issues that people might have had in bare workflow because unimodules were not specyfing dependencies' versions.

# Test Plan

- [x] Xcode builds the project.
- [x] `et add-sdk -p ios -s 36.0.0` generated correct `dependencies.rb` and `postinstalls.rb` scripts and they were correctly evaluated by `pod install`
- [x] `et rm-sdk -p ios -s 36.0.0` worked as well.
- [x] `client_ios` CI job is green

ide committed:
[home] Remove deprecated componentWillMount calls

When these were performing work, moved that work into componentDidMount. When they weren't, moved the code into the constructor. This isn't always safe but this time it was straightforward.

tsapeta committed:
[workspaces] fix some dependencies not being symlinked

tsapeta commented on packages/expo-local-authentication/src/LocalAuthentication.ts:
```diff
@@ -48,9 +48,12 @@ export async function authenticateAsync(
 
   if (Platform.OS === 'ios') {
+    if (!options.promptMessage) {
+      options.promptMessage = 'Authenticate';
```
I'm not a big fan of modifying an object that was passed as an argument to the function. This may cause some unexpected behavior for the user who use that object also for something else.
It might be better to do sth like `const promptMessage = options.promptMessage || 'Authenticate';` on top of the function, below the if block with UnavailabilityError 😉 
And then call the native method in that way:
```javascript
await ExpoLocalAuthentication.authenticateAsync({
  ...options,
  promptMessage,
});
```

lukmccall committed:
[media-library] Fix getAssetInfoAsync doesn't return localUri for videos (#5806)

# Why

Resolve #2448.

# How

For video files, to get additional information we need to use `requestAVAssetForVideo` method.

# Test Plan

- https://snack.expo.io/@lukaszkosmaty/medialibrary.getassetinfoasync-does-not-return-localuri-for-mediatype-%22video%22 - now display `local Uri` property for both images and videos.

cruzach committed:
feedback: do not modify input object

ide commented on .gitignore:
```diff
@@ -75,7 +75,6 @@ jarjar-rules.txt
 /android/app/fabric.properties
 /ios/Exponent/Generated/EXKeys.h
 /ios/ExponentIntegrationTests/EXTestEnvironment.plist
-/ios/Podfile
```
revert all this, your git branch looks like it is in a bad state

cruzach replied:
yeah, made a mistake and rebased from master

cruzach committed:
revert bad rebase
